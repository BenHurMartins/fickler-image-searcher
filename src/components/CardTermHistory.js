import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {Colors, Mixins} from '../styles';

const CardTermHistory = ({term, onPress}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={() => onPress(term)}>
      <Text style={styles.term}>{term}</Text>
    </TouchableOpacity>
  );
};

export default CardTermHistory;

const styles = StyleSheet.create({
  container: {
    width: Mixins.DEVICE_WIDTH_90,
    justifyContent: 'center',
    backgroundColor: Colors.OFF_WHITE,
    paddingLeft: 10,
    height: 40,
    zIndex: 9999,
  },
  term: {
    fontSize: Mixins.FONT_SIZE_REGULAR,
    color: Colors.BLACK,
  },
});
