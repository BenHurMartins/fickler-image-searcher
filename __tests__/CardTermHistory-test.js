import 'react-native';
import React from 'react';
import {render, act} from '@testing-library/react-native';
import CardTermHistory from '../src/components/CardTermHistory';
import renderer from 'react-test-renderer';

describe('CardTermHistory component', () => {
  it('CardTermHistory renders correctly', () => {
    renderer.create(<CardTermHistory onPress={() => null} term={'Test'} />);
  });
  it('CardTermHistory snapshot test', () => {
    const tree = renderer
      .create(<CardTermHistory onPress={() => null} term={'Test'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should set the term', () => {
    const {getByText} = render(
      <CardTermHistory onPress={() => null} term={'Test'} />,
    );
    act(() => {
      expect(getByText('Test')).not.toBe(null);
    });
  });
});
