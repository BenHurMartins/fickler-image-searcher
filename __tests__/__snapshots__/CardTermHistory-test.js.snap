// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTermHistory component CardTermHistory snapshot test 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "#F0F0F0",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 10,
        "width": 675,
        "zIndex": 9999,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 30,
        }
      }
    >
      Test
    </Text>
  </View>
</RNGestureHandlerButton>
`;
